packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -535,11 +537,13 @@ export class Sha256 extends Op {
    * Asserts 0 arguments are passed.
    * @param args Expected arguments: [] // none
    * @param line line number in TEAL file
+   * @param interpreter interpreter object
    */
-  constructor (args: string[], line: number) {
+  constructor (args: string[], line: number, interpreter: Interpreter) {
     super();
     this.line = line;
     assertLen(args.length, 0, line);
+    interpreter.gas += interpreter.tealVersion === 1 ? 7 : 35;
```

Comment:
can we use a map for gas ? Ex: version->opcodes-> gas (maybe two maps one for version1 and other for 2).
- They are more clear
- Easily maintainable(changing few according to current code is tedious task)
- Also values are constant, so they are unrelated to opcode logic.
what are the benefits of embedding it in constructor ?

Reply:
Yes. Also, only crypto ops have different values for v1 and v2. Rest all the opcodes have a gas cost of 1. So with them in constructor, we could just do `interpreter.gas++`, after verifying min version. We can create a 2 maps for crypto ops (v1, v2).

Reply:
Is it possible to move calculation out of constructor ? This way updating will be easier, let's say version 3 comes.

We can have one function which will add up gas using map, and store it, without going to each opcode constructor.

Reply:
We can take the calculation out of the constructor by moving gas cost calculation in the parser itself. I am not sure about using maps though, as most of the opcodes have cost 1 in both version. Maybe use a Set?

Reply:
Yes we can use set. Or check for opcode in map(it will only have opcodes that has cost other than 1).  If it is not in map we can add 1. We can put this logic in parser where we already checked if opcodes is known.

Reply:
Ok